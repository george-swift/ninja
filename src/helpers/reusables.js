const randomInt = (min, max) => Math.floor(Math.random() * (max - min + 1)) + min;

const inputValidator = (field) => {
  const rex = /\d+|\w+/;
  return !!rex.exec(field.value);
};

const checkField = (elem, className) => {
  if (!elem.classList.contains(className)) return;
  elem.classList.remove(className);
};

const eltBuilder = (name, attrs, ...children) => {
  const dom = document.createElement(name);

  Object.entries(attrs).forEach(([key, value]) => {
    dom.setAttribute(key, value);
  });

  [...children].forEach((child) => dom.appendChild(child));

  return dom;
};

const storeKata = (kata) => {
  const total = JSON.stringify(kata);
  localStorage.setItem('kata', total);
};

const retrieveKata = () => {
  const total = localStorage.getItem('kata') ? JSON.parse(localStorage.getItem('kata')) : 0;
  return total;
};

export {
  randomInt,
  inputValidator,
  checkField,
  eltBuilder,
  storeKata,
  retrieveKata,
};