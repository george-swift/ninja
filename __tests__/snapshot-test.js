/**
 * @jest-environment jsdom
*/

import { eltBuilder } from '../src/helpers/reusables.js';

describe('Snapshot testing for consistent UI design', () => {
  test('Element builder does not change unexpectedly', () => {
    const templateContainer = eltBuilder('div', {
      class: 'shadow-lg p-4',
      id: 'phaser-game',
    }, eltBuilder('div', {
      class: 'canvas-sibling',
    }), eltBuilder('canvas', {
      id: 'canvas',
    }));
    expect(templateContainer).toMatchSnapshot();
  });
});