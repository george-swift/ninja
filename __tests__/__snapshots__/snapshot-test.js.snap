// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for consistent UI design Element builder does not change unexpectedly 1`] = `
<div
  class="shadow-lg p-4"
  id="phaser-game"
>
  <div
    class="canvas-sibling"
  />
  <canvas
    id="canvas"
  />
</div>
`;
